NoSuchMethodException: java/lang/invoke/MethodHandle.invoke on x86

Extended system tests are failing on x86

https://ci.eclipse.org/openj9/job/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/2
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK8-linux_x86-64_cmprssptrs/1/
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK8-linux_x86-64/2/
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK10-linux_x86-64/3/

```
===============================================
Running test LangLoadTest_0 ...
===============================================
LangLoadTest_0 Start Time: Thu Aug  2 04:55:14 2018 Epoch Time (ms): 1533185714522
test with NoOptions
STF 04:55:14.693 - =========================   S T F   =========================
systemtest-prereqs has been processed, and set to: /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqsRetrieving amount of free space on drive containing /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0
There is 90387 Mb free
STF 04:55:14.699 - ==================   G E N E R A T I O N   ==================
STF 04:55:14.700 - Checking JVM: /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/bin/../
STF 04:55:14.700 - Starting process to generate scripts: /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/bin/..//bin/java  -Dlog4j.skipJansi=true -Djava.system.class.loader=net.adoptopenjdk.stf.runner.StfClassLoader -classpath /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-api-2.3.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-core-2.3.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.core/scripts/../bin net.adoptopenjdk.stf.runner.StfRunner -properties "/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/stf_parameters.properties, , /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.core/config/stf.properties" -testDir "/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest"
GEN 04:55:15.909 - Found test. Project: 'openjdk.test.load' class: 'LangLoadTest.class' Dir: '/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/openjdk-systemtest/openjdk.test.load/bin'
GEN 04:55:15.912 - Found test. Project: 'openjdk.test.load' class: 'net.adoptopenjdk.stf.LangLoadTest'
GEN Classpath directories used by project 'openjdk.test.load': 
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/openjdk-systemtest/openjdk.test.load/bin
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.core/bin
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.load/bin
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-api-2.3.jar
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-core-2.3.jar
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/junit-4.12/junit-4.12.jar
GEN   /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/junit-4.12/hamcrest-core-1.3.jar
GEN 04:55:16.241 - Using Mode NoOptions. Values = ''
GEN 04:55:16.330 - 
GEN 04:55:16.330 - Test command summary:
GEN 04:55:16.331 -   Step  Stage   Command           Description
GEN 04:55:16.331 -  -----+--------+-----------------+------------
GEN 04:55:16.331 -     1  execute  Run java          Run lang load test
STF 04:55:16.377 - 
STF 04:55:16.377 - Script generation completed
STF 04:55:16.377 - 
STF 04:55:16.377 - 
STF 04:55:16.377 - =======================   S E T U P   =======================
STF 04:55:16.377 - Running setup: perl /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/setUp.pl
STF 04:55:16.442 - SETUP stage completed
STF 04:55:16.446 - 
STF 04:55:16.446 - ====================   E X E C U T E -   ====================
STF 04:55:16.446 - Running execute: perl /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/execute.pl
STF 04:55:16.512 - 
STF 04:55:16.512 - Java version
STF 04:55:16.512 - Running: /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/bin/../bin/java -version
openjdk version "10.0.2-internal" 2018-07-17
OpenJDK Runtime Environment (build 10.0.2-internal+0-adhoc.jenkins.Build-JDK10-linuxx86-64cmprssptrs)
Eclipse OpenJ9 VM (build master-1517aa5, JRE 10 Linux amd64-64-Bit Compressed References 20180802_313 (JIT enabled, AOT enabled)
OpenJ9   - 1517aa5
OMR      - b971ea0
JCL      - 01b7e8c based on jdk-10.0.2+13)
STF 04:55:16.714 - 
STF 04:55:16.714 - +------ Step 1 - Run lang load test
STF 04:55:16.714 - | Run foreground process
STF 04:55:16.714 - |   Program:     /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/bin/../bin/java
STF 04:55:16.714 - |   Mnemonic:    LLT
STF 04:55:16.714 - |   Echo:        ECHO_ON
STF 04:55:16.714 - |   Expectation: CLEAN_RUN within 10m
STF 04:55:16.714 - |
STF 04:55:16.714 - Running command: /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/bin/../bin/java -Xcompressedrefs -classpath /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.load/bin:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/stf/stf.core/bin:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-api-2.3.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/log4j-2.3/log4j-core-2.3.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/junit-4.12/junit-4.12.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/systemtest_prereqs/junit-4.12/hamcrest-core-1.3.jar:/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/jvmtest/systemtest/openjdk-systemtest/openjdk.test.lang/bin net.adoptopenjdk.loadTest.LoadTest -resultsDir /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results -resultsPrefix 1.LLT. -reportFailureLimit 1 -abortAtFailureLimit 10 -maxTotalLogFileSpace 200M -maxSingleLogSize 1/25 -suite.lang.threadCount 3 -suite.lang.totalNumberTests 3100 -suite.lang.inventoryFile /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results/1.LLT.inventory/openjdk.test.load/config/inventories/lang/lang.xml -suite.lang.inventoryExcludeFile none -suite.lang.selection random -suite.lang.seed -1 -suite.lang.repeatCount 1 -suite.lang.thinkingTime 0ms..0ms
STF 04:55:16.714 - Redirecting stderr to /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results/1.LLT.stderr
STF 04:55:16.714 - Redirecting stdout to /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results/1.LLT.stdout
STF 04:55:16.725 - Monitoring processes: LLT
LLT 04:55:17.990 - Load test parameters
LLT 04:55:17.992 -   Time limited         = false
LLT 04:55:17.993 -   abortIfOutOfMemory   = true
LLT 04:55:17.993 -   reportFailureLimit   = 1
LLT 04:55:17.994 -   abortAtFailureLimit  = 10
LLT 04:55:17.995 -   maxTotalLogFileSpace = 209715200
LLT 04:55:17.996 -   maxSingleLogSize     = 8388608
LLT 04:55:17.996 - Parameters for suite 0
LLT 04:55:17.997 -   Suite name     = lang
LLT 04:55:17.998 -   Number threads = 3
LLT 04:55:17.998 -   Supplied seed  = -1
LLT 04:55:17.999 -   Inventory file = /home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results/1.LLT.inventory/openjdk.test.load/config/inventories/lang/lang.xml
LLT 04:55:18.000 -   Exclude file   = none
LLT 04:55:18.001 -   Number tests   = 3100
LLT 04:55:18.001 -   Repeat count   = 1
LLT 04:55:18.002 -   Thinking time  = 0ms..0ms
LLT 04:55:18.003 -   Selection mode = random
LLT 04:55:18.004 -   Actual seed    = 1532926918991
LLT 04:55:18.013 - Parsing inventory file. Root=/home/jenkins/workspace/Test-extended.system-JDK10-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15331856985700/LangLoadTest_0/20180802-045514-LangLoadTest/results/1.LLT.inventory File=openjdk.test.load/config/inventories/lang/lang.xml
LLT 04:55:18.062 - Final test list:
LLT 04:55:18.065 -   0 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicTest doTest]  Weighting=1 
LLT 04:55:18.065 -   1 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicTest2 runAllTests]  Weighting=1 
LLT 04:55:18.065 -   2 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.StaticAsTypeTestExplicit testAll]  Weighting=1 
LLT 04:55:18.065 -   3 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.StaticAsTypeTestImplicit testAll]  Weighting=1 
LLT 04:55:18.065 -   4 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testVoid]  Weighting=1 
LLT 04:55:18.066 -   5 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testVoid]  Weighting=1 
LLT 04:55:18.066 -   6 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testReference]  Weighting=1 
LLT 04:55:18.066 -   7 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testReference]  Weighting=1 
LLT 04:55:18.066 -   8 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testByte]  Weighting=1 
LLT 04:55:18.066 -   9 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testByte]  Weighting=1 
LLT 04:55:18.067 -   10 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testBoolean]  Weighting=1 
LLT 04:55:18.067 -   11 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testBoolean]  Weighting=1 
LLT 04:55:18.067 -   12 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testShort]  Weighting=1 
LLT 04:55:18.067 -   13 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testShort]  Weighting=1 
LLT 04:55:18.068 -   14 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testChar]  Weighting=1 
LLT 04:55:18.068 -   15 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testChar]  Weighting=1 
LLT 04:55:18.069 -   16 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testInt]  Weighting=1 
LLT 04:55:18.069 -   17 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testInt]  Weighting=1 
LLT 04:55:18.070 -   18 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testLong]  Weighting=1 
LLT 04:55:18.070 -   19 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testLong]  Weighting=1 
LLT 04:55:18.071 -   20 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testFloat]  Weighting=1 
LLT 04:55:18.072 -   21 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testFloat]  Weighting=1 
LLT 04:55:18.072 -   22 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testDouble]  Weighting=1 
LLT 04:55:18.073 -   23 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.AsTypeTest testDouble]  Weighting=1 
LLT 04:55:18.073 -   24 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicStaticTest testStaticSetter]  Weighting=1 
LLT 04:55:18.074 -   25 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicStaticTest testStaticGetter]  Weighting=1 
LLT 04:55:18.074 -   26 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicStaticTest testStaticStringSetter]  Weighting=1 
LLT 04:55:18.074 -   27 ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicStaticTest testStaticStringGetter]  Weighting=1 
LLT 04:55:18.075 -   28 JUnit[net.adoptopenjdk.test.lang.java7.ProjectCoin.SwitchOnString]  Weighting=1 
LLT 04:55:18.076 -   29 JUnit[net.adoptopenjdk.test.lang.java7.ProjectCoin.TryWithResources]  Weighting=1 
LLT 04:55:18.076 -   30 JUnit[net.adoptopenjdk.test.lang.java7.ProjectCoin.MultiCatchPreciseRethrow]  Weighting=1 
LLT 04:55:18.124 - Starting thread. Suite=0 thread=0
LLT 04:55:18.125 - Starting thread. Suite=0 thread=1
LLT 04:55:18.125 - Starting thread. Suite=0 thread=2
LLT 04:55:21.107 - First failure detected by thread: load-0. Not creating dumps as not running on an IBM JVM
LLT 04:55:21.111 - Test failed
LLT   Failure num.  = 1
LLT   Test number   = 1
LLT   Test details  = 'ArbitraryJava[net.adoptopenjdk.test.lang.invoke.BasicTest2 runAllTests]'
LLT   Suite number  = 0
LLT   Thread number = 0
LLT >>> Captured test output >>>
LLT testInvokeHandle : Test failed:
LLT java.lang.reflect.InvocationTargetException
LLT 	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
LLT 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@10.0.2-internal/DelegatingMethodAccessorImpl.java:43)
LLT 	at java.lang.reflect.Method.invoke(java.base@10.0.2-internal/Method.java:564)
LLT 	at net.adoptopenjdk.test.lang.invoke.BasicTest2.runTest(BasicTest2.java:90)
LLT 	at net.adoptopenjdk.test.lang.invoke.BasicTest2.runTests(BasicTest2.java:97)
LLT 	at net.adoptopenjdk.test.lang.invoke.BasicTest2.runAllTests(BasicTest2.java:116)
LLT 	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
LLT 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@10.0.2-internal/DelegatingMethodAccessorImpl.java:43)
LLT 	at java.lang.reflect.Method.invoke(java.base@10.0.2-internal/Method.java:564)
LLT 	at net.adoptopenjdk.loadTest.adaptors.ArbitraryJavaAdaptor.executeTest(ArbitraryJavaAdaptor.java:101)
LLT 	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:177)
LLT 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@10.0.2-internal/ThreadPoolExecutor.java:1135)
LLT 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@10.0.2-internal/ThreadPoolExecutor.java:635)
LLT 	at java.lang.Thread.run(java.base@10.0.2-internal/Thread.java:835)
LLT Caused by: java.lang.NoSuchMethodException: java/lang/invoke/MethodHandle.invoke(Lnet/adoptopenjdk/test/lang/invoke/BasicTest2$ArrayTaker;FF[J)I (loaded from jrt:/java.base by <Bootstrap Loader>) called from class java.lang.invoke.MethodHandle (loaded from jrt:/java.base by <Bootstrap Loader>).
LLT 	at java.lang.invoke.PrimitiveHandle.finishMethodInitialization(java.base@10.0.2-internal/PrimitiveHandle.java:194)
LLT 	at java.lang.invoke.DirectHandle.<init>(java.base@10.0.2-internal/DirectHandle.java:51)
LLT 	at java.lang.invoke.MethodHandles$Lookup.findVirtual(java.base@10.0.2-internal/MethodHandles.java:620)
LLT 	at net.adoptopenjdk.test.lang.invoke.BasicTest2.testInvokeHandle(BasicTest2.java:276)
LLT 	... 14 more
```

--- a/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
+++ b/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
@@ -4715,7 +4715,7 @@ inline void generateInlinedCheckCastOrInstanceOfForClass(TR::Node* node, TR_Opaq
    generateLoadJ9Class(node, j9class, j9class, cg);
 
    // Equality test
-   if (!fej9->isAbstractClass(clazz))
+   if (!fej9->isAbstractClass(clazz) || node->getOpCodeValue() == TR::icall/*TR_checkAssignable*/)
       {
       if (use64BitClasses)
          {

